includes/Checker/Checks/Plugin_Repo/Plugin_Header_Fields_Check.php

```diff
@@ -0,0 +1,335 @@
+	 *
+	 * @since 1.2.0
+	 *
+	 * @return array The categories for the check.
+	 */
+	public function get_categories() {
+		return array( Check_Categories::CATEGORY_PLUGIN_REPO );
+	}
+
+	/**
+	 * Amends the given result by running the check on the associated plugin.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 *
+	 * @throws Exception Thrown when the check fails with a critical error (unrelated to any errors detected as part of the check).
+	 *
+	 * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+	 * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 */
+	public function run( Check_Result $result ) {
+		if ( ! function_exists( 'get_plugin_data' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		$plugin_main_file = $result->plugin()->main_file();
+		$plugin_header    = get_plugin_data( $plugin_main_file );
+
+		$labels = array(
+			'Name'            => 'Plugin Name',
+			'PluginURI'       => 'Plugin URI',
+			'Version'         => 'Version',
+			'Description'     => 'Description',
+			'Author'          => 'Author',
+			'AuthorURI'       => 'Author URI',
+			'TextDomain'      => 'Text Domain',
+			'DomainPath'      => 'Domain Path',
+			'Network'         => 'Network',
+			'RequiresWP'      => 'Requires at least',
+			'RequiresPHP'     => 'Requires PHP',
+			'UpdateURI'       => 'Update URI',
+			'RequiresPlugins' => 'Requires Plugins',
+		);
+
+		if ( ! empty( $plugin_header['Name'] ) ) {
+			if ( in_array( $plugin_header['Name'], array( 'Plugin Name', 'My Basics Plugin' ), true ) ) {
+				$this->add_result_warning_for_file(
+					$result,
+					sprintf(
+						/* translators: %s: plugin header field */
+						__( 'The "%s" header in the plugin file is not valid.', 'plugin-check' ),
+						esc_html( $labels['Name'] )
+					),
+					'plugin_header_invalid_plugin_name',
+					$plugin_main_file,
+					0,
+					0,
+					'',
+					6
+				);
+			}
+		}
+
+		if ( ! empty( $plugin_header['PluginURI'] ) ) {
+			if ( true !== $this->is_valid_url( $plugin_header['PluginURI'] ) ) {
+				$this->add_result_warning_for_file(
+					$result,
+					sprintf(
+						/* translators: %s: plugin header field */
+						__( 'The "%s" header in the plugin file is not valid.', 'plugin-check' ),
+						esc_html( $labels['PluginURI'] )
+					),
+					'plugin_header_invalid_plugin_uri',
+					$plugin_main_file,
+					0,
+					0,
+					'',
+					6
+				);
+			} elseif ( str_contains( $plugin_header['PluginURI'], '//wordpress.org/' ) || str_contains( $plugin_header['PluginURI'], '//example.com/' ) ) {
+				$this->add_result_warning_for_file(
+					$result,
+					sprintf(
+						/* translators: %s: plugin header field */
+						__( 'The "%s" header in the plugin file uses restricted domain.', 'plugin-check' ),
+						esc_html( $labels['PluginURI'] )
+					),
+					'plugin_header_restricted_plugin_uri',
+					$plugin_main_file,
```

Comment:
Not sure "restricted" is the right word.

We could just use the same error message as above, or say it's not "allowed".

Curious to hear what others think.

Reply:
I'm agree with Pascal. I propose to say It's not "valid".